File: app/src/main/java/in/nisb/nisbapp/MessageService.java
package in.nisb.nisbapp;

import android.content.Intent;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import org.json.JSONException;
import org.json.JSONObject;


public class MessageService extends FirebaseMessagingService {

    public static String eventid="";

    public MessageService() {
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {


        Intent intent = new Intent(getApplicationContext(), MainActivity.class);
        try {
            JSONObject jo = new JSONObject();
            jo.put("context", remoteMessage.getData().get("context"));
            jo.put("body",remoteMessage.getData().get("body"));

            if (remoteMessage.getData().get("context").equals("event-single")) {
                intent = new Intent(getApplicationContext(), EventSingleActivity.class);
                intent.putExtra("id", remoteMessage.getData().get("id"));
                jo.put("id",remoteMessage.getData().get("id"));
            }


            if (remoteMessage.getData().get("context").equals("blog-single")) {
                intent = new Intent(getApplicationContext(), BlogSingleActivity.class);
                intent.putExtra("content", remoteMessage.getData().get("content"));
                intent.putExtra("title", remoteMessage.getData().get("title"));
                intent.putExtra("extra", remoteMessage.getData().get("extra"));
                jo.put("content",remoteMessage.getData().get("content"));
                jo.put("title",remoteMessage.getData().get("title"));
                jo.put("extra",remoteMessage.getData().get("extra"));
            }

            NisbUser.addNotification(getApplicationContext(),jo.toString());
        }
        catch (JSONException j){

        }
        ExtraFunctions.sendNotification(getApplicationContext(),
                "NISB",
                remoteMessage.getData().get("body"),intent);
    }

}

File: app/src/main/java/in/nisb/nisbapp/services/MessageService.kt
package `in`.nisb.nisbapp

import android.content.Intent

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

import org.json.JSONException
import org.json.JSONObject


class MessageService : FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage?) {

        var intent = Intent(applicationContext, MainActivity::class.java)
        try {
            val jo = JSONObject()
            jo.put("context", remoteMessage!!.data["context"])
            jo.put("body", remoteMessage.data["body"])

            if (remoteMessage.data["context"] == "event-single") {
                intent = Intent(applicationContext, EventSingleActivity::class.java)
                intent.putExtra("id", remoteMessage.data["id"])
                jo.put("id", remoteMessage.data["id"])
            }


            if (remoteMessage.data["context"] == "blog-single") {
                intent = Intent(applicationContext, BlogSingleActivity::class.java)
                intent.putExtra("url", remoteMessage.data["blogurl"])
                jo.put("url", remoteMessage.data["blogurl"])
            }

            //Save the info to the DB
            DatabaseFunctions().addNotification(applicationContext, jo.toString())
        } catch (j: JSONException) {

        }

        //Show the Notification
        ExtraFunctions.sendNotification( applicationContext,
                "NISB", remoteMessage!!.data["body"], intent)
    }

}
